src/components/AdrenalineConnector.jsx

```diff
@@ -46,7 +46,8 @@ export default class AdrenalineConnector extends Component {
 
   hasSliceChanged(props, slice, nextSlice) {
     const { adrenaline} = props;
-    return adrenaline.hasStateChanged((slice||{}).props, (nextSlice||{}).props);
+    return !shallowEqual((slice || {}).variables, (nextSlice || {}).variables) ||
```

Comment:
I think the better option here is to set defaults for `slice` and `nextSlice` in the method signature.